src/main/java/org/opentripplanner/middleware/utils/HttpUtils.java

```diff
@@ -103,12 +104,35 @@ public static String httpRequest(URI uri, int connectionTimeout, REQUEST_METHOD
         return null;
     }
 
+    public static int getQueryParamFromRequest(Request req, String name, boolean optional, int min, int defaultValue, int max) {
+        // Start with default value
+        int value = defaultValue;
+        String requestValue = null;
+        try {
+            // Attempt to get value from query param.
+            requestValue = HttpUtils.getQueryParamFromRequest(req, name, optional);
+            if (requestValue != null) {
+                value = Integer.parseInt(requestValue);
+                // If requested value is out of range, pin to min/max.
+                if (value < min) value = min;
+                else if (max != -1 && value > max) value = max;
```

Comment:
To specify no limit on the max value, something like `Integer.MAX_VALUE` or `Double.POSITIVE_INFINITY` should be used. Since -1 is a real number, this design could introduce a bug if -1 is actually an acceptable maximum value.